src/main/scala/io/iohk/ethereum/ledger/BlockPreparator.scala

```diff
@@ -43,17 +43,32 @@ class BlockPreparator(
     def getAccountToPay(address: Address, ws: InMemoryWorldStateProxy): Account = ws.getAccount(address)
       .getOrElse(Account.empty(blockchainConfig.accountStartNonce))
 
+    val blockNumber = block.header.number
+    // condition to calculate block rewards https://github.com/ethereum/EIPs/blob/master/EIPS/eip-649.md
+    val shouldReduceBlockReward = blockNumber >= blockchainConfig.byzantiumBlockNumber
+
     val minerAddress = Address(block.header.beneficiary)
     val minerAccount = getAccountToPay(minerAddress, worldStateProxy)
-    val minerReward = blockRewardCalculator.calcBlockMinerReward(block.header.number, block.body.uncleNodesList.size)
+    val minerReward = if (shouldReduceBlockReward) {
+      BigInt(10).pow(18) * 3
```

Comment:
I think all changes should be localized to class `BlockRewardCalculator`.
Specs are sayin that: `This is the existing pre-Metropolis formula for uncle rewards, simply adjusted with new_block_reward` , so i think all our current formulas for rewards in https://github.com/input-output-hk/mantis/blob/feature/byzantium_fork/src/main/scala/io/iohk/ethereum/ledger/BlockRewardCalculator.scala are correct, and only change we would need to do there is to change value https://github.com/input-output-hk/mantis/blob/feature/byzantium_fork/src/main/scala/io/iohk/ethereum/ledger/BlockRewardCalculator.scala#L19 to `3,000,000,000,000,000,000` if we are past byzantium block. (it can be done here, or mabe to config passed to calculator can be updated if we are past byzantium block)

Reply:
I tried this, but when I applied it, there were more errors than in the real solution, so I just moved all the calculations to the calculator class